camera_alignment_core/alignment_utils/alignment_qc.py

```diff
@@ -0,0 +1,363 @@
+
+        if len(missing) == 0:
+            return None
+        else:
+            return missing
+
+    # QC Functions ############################################################
+
+    def check_z_offset_between_ref_mov(self) -> Tuple[int, int, int]:
+        if self.reference is None or self.moving_source is None:
+            log.error("Error: Raw images are missing for qc")
+            raise Exception("Error: Raw images are missing for qc")
+            return -1, -1, -1
+
+        z_offset = self.ref_origin - self.mov_origin
+
+        return z_offset, self.ref_origin, self.mov_origin
+
+    def report_ref_mov_image_snr(self) -> Tuple[int, int, int, int]:
+        if self.reference is None or self.moving_source is None:
+            log.error("Error: Seg images are missing for qc")
+            raise Exception("Error: Seg images are missing for qc")
+            return -1, -1, -1, -1
+
+        def get_image_snr(
+            seg: Optional[NDArray[np.uint16]],
+            img_intensity: Optional[NDArray[np.uint16]],
+        ) -> Tuple[int, int]:
+            if seg is None or img_intensity is None:
+                return -1, -1
+            signal = np.median(img_intensity[seg.astype(bool)])
+            noise = np.median(img_intensity[~seg.astype(bool)])
+
+            return signal, noise
+
+        ref_signal, ref_noise = get_image_snr(
+            self.reference_seg, self.reference[self.ref_origin]
+        )
+        mov_signal, mov_noise = get_image_snr(
+            self.moving_seg, self.moving_source[self.mov_origin]
+        )
+
+        return ref_signal, ref_noise, mov_signal, mov_noise
+
+    def report_number_beads(self) -> Tuple[bool, int]:
+        if self.ref_mov_coor_dict is None:
+            log.error("Error: ref_mov_coor_dict is missing for qc")
+            raise Exception("Error: ref_mov_coor_dict is missing for qc")
+            return False, -1
+
+        bead_num_qc = False
+        num_beads = len(self.ref_mov_coor_dict)
+        if num_beads >= 10:
+            bead_num_qc = True
+        return bead_num_qc, num_beads
+
+    def report_change_fov_intensity_parameters(self) -> Dict[str, int]:
+        """
+        Reports changes in FOV intensity after transform
+        :return: A dictionary with the following keys and values:
+            median_intensity
+            min_intensity
+            max_intensity
+            1_percentile: first percentile intensity
+            995th_percentile: 99.5th percentile intensity
+        """
+        if self.moving_transformed is None or self.moving_source is None:
+            log.error("Error: moving source and transformed images are missing for qc")
+            raise Exception(
+                "Error: moving source and transformed images are missing for qc"
+            )
+            return {"ERROR": -1}
+
+        change_fov_intensity_param_dict = {
+            "median_intensity": np.median(self.moving_transformed) * 65535
```

Comment:
Maybe 65535 could be a CONSTANT?  or maybe it is an obvious number to relevant people with more context?

Reply:
I believe this was determined by others for whom it's relevant